framework/src/relationshipmanagers/GhostBoundary.C

```diff
@@ -0,0 +1,135 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// App includes
+#include "GhostBoundary.h"
+#include "Executioner.h"
+#include "FEProblemBase.h"
+#include "MooseApp.h"
+
+// libMesh includes
+#include "libmesh/elem.h"
+#include "libmesh/mesh_base.h"
+#include "libmesh/boundary_info.h"
+
+registerMooseObject("MooseApp", GhostBoundary);
+
+using namespace libMesh;
+
+InputParameters
+GhostBoundary::validParams()
+{
+  InputParameters params = RelationshipManager::validParams();
+  params.addRequiredParam<std::vector<BoundaryName>>("boundary",
+                                                     "The name of the primary boundary sideset.");
+
+  params.addParam<bool>("ghost_point_neighbors",
+                        false,
+                        "Whether we should ghost point neighbors of the paired boundary.");
+
+  // We want to wait until our mortar mesh has been built before trying to delete remote elements.
+  // And our mortar mesh cannot be built until the entire mesh has been generated. By setting this
+  // parameter to false we will make sure that any prepare_for_use calls during the mesh generation
+  // phase will not delete remote elements *and* we will set a flag on the moose mesh saying that we
+  // need to delete remote elements after the addition of late geometric ghosting functors
+  // (including this ghosting functor)
```

Comment:
```suggestion
```

Reply:
I think we still need `ghost_point_neighbors` for this operator `virtual bool operator>=(const RelationshipManager & other) const override;` or am I missing something?

Reply:
If we don't use it in determining ghosting, then it shouldn't be in the class, which means it shouldn't be used in the evaluation of `>=`

Reply:
Not sure about this one, it is not used in setting the ghosting, but it is in checking the `>=` , as in
`if (_boundary_name == asoi->_boundary_name &&
        _ghost_point_neighbors >= asoi->_ghost_point_neighbors)
      return true;`
and this operator seems to be relevant in other parts of the code I have no overview of. 
I am all for rewriting this check, since otherwise `ghost_boundary_points` is not needed,  but I have no reference point for it.

Reply:
All you have to do is change to `if (_boundary_name == asoi->_boundary_name) return true;`. This is saying that two given `GhostBoundary` relationship managers are equivalent if they are ghosting the same boundaries. It would be be a misnomer to return `false` if the same boundaries are being ghosted but the values of `ghost_point_neighbors` are different because the latter has zero impact on the number and location of elements ghosted

Reply:
If we really wanted to be accurate in this comparison we would be checking to see whether `asoi->_boundary_name` is equivalent *or* a subset of `_boundary_name`, but I'm not going to require that change